apps/evm/lib/evm/vm.ex

```diff
@@ -144,4 +144,41 @@ defmodule EVM.VM do
 
     {final_machine_state, n_sub_state, n_exec_env}
   end
+
+  def log(operation, machine_state) do
+    operation_string = Atom.to_string(operation.sym)
+      |> String.upcase()
+      |> String.pad_leading(8)
+
+    opcode_string = operation.id
+      |> :binary.encode_unsigned()
+      |> Base.encode16(case: :lower)
+      |> String.trim_leading("0")
+      |> String.pad_trailing(2)
+
+    program_counter = machine_state.program_counter + 1
+    program_counter_string = program_counter
```

Comment:
Are you using the code formatter? I believe it would format these by putting `program_counter` on the next line, 

```
program_counter_string = 
  program_counter
  |> :binary.encode_unsigned()
# rest
```